src/__tests__/deployments.helper.ts

```diff
@@ -0,0 +1,126 @@
+    'CompatibilityFallbackHandler', // 1.3.0, 1.4.1
+  ],
+  CreateCall: [
+    'CreateCall', // 1.3.0, 1.4.1
+  ],
+  DefaultCallbackHandler: [
+    'DefaultCallbackHandler', // 1.1.1
+  ],
+  MultiSend: [
+    'MultiSend', // 1.3.0, 1.4.1
+  ],
+  MultiSendCallOnly: [
+    'MultiSendCallOnly', // 1.3.0, 1.4.1
+  ],
+  ProxyFactory: [
+    'ProxyFactory', // 1.0.0, 1.1.1
+    'GnosisSafeProxyFactory', // 1.3.0
+    'SafeProxyFactory', // 1.4.1
+  ],
+  Safe: [
+    'GnosisSafe', // 1.0.0, 1.1.1, 1.2.0, 1.3.0
+    'Safe', // 1.4.1
+  ],
+  SafeL2: [
+    'GnosisSafeL2', // 1.3.0
+    'SafeL2', // 1.4.1
+  ],
+  SignMessageLib: [
+    'SignMessageLib', // 1.3.0, 1.4.1
+  ],
+  SimulateTxAccessor: [
+    'SimulateTxAccessor', // 1.3.0, 1.4.1
+  ],
+};
+
+// Path to directory containing JSON assets
+const assetsDir = path.join(
+  process.cwd(),
+  'node_modules',
+  '@safe-global',
+  'safe-deployments',
+  'dist',
+  'assets',
+);
+
+type VersionsByChainIdByDeploymentMap = {
+  [contractAlias: string]: { [chainId: string]: string[] | undefined };
+};
+
+let versionsByDeploymentByChainIdSingleton: VersionsByChainIdByDeploymentMap | null;
+
+export function getVersionsByChainIdByDeploymentMap(): VersionsByChainIdByDeploymentMap {
+  // If previously computed, return the singleton
+  if (versionsByDeploymentByChainIdSingleton) {
+    return versionsByDeploymentByChainIdSingleton;
+  }
+
+  // Initialize map
+  versionsByDeploymentByChainIdSingleton = {};
+  for (const contractName of Object.keys(deploymentAliases)) {
+    versionsByDeploymentByChainIdSingleton[contractName] = {};
+  }
+
+  // For each version...
+  for (const version of fs.readdirSync(assetsDir)) {
+    const versionDir = path.join(assetsDir, version);
+
+    // ...parse each asset
+    for (const assetFile of fs.readdirSync(versionDir)) {
+      // Read the asset JSON
+      const assetPath = path.join(assetsDir, version, assetFile);
+      const assetJson = fs.readFileSync(assetPath, 'utf8');
+
+      // Parse the asset JSON
+      const deployment = JSON.parse(assetJson);
+
+      // Get the alias name
+      const name = Object.entries(deploymentAliases).find(([, aliases]) => {
+        return aliases.includes(deployment.contractName);
+      })?.[0];
+
+      if (!name) {
+        throw new Error(
+          `The ${deployment.contractName} contract alias is not known!`,
+        );
+      }
+
+      // Add the version to the map
+      for (const chainId of Object.keys(deployment.networkAddresses)) {
+        versionsByDeploymentByChainIdSingleton[name][chainId] ??= [];
+        versionsByDeploymentByChainIdSingleton[name][chainId]?.push(
+          deployment.version,
+        );
+      }
+    }
+  }
+
+  return versionsByDeploymentByChainIdSingleton;
+}
+
+export function getDeploymentVersionsByChainIds(
+  contractAlias: keyof typeof deploymentAliases,
+  chainIds: Array<string>,
+): Record<string, Array<string>> {
+  const map = getVersionsByChainIdByDeploymentMap();
```

Comment:
I think we should store this `map` on a global scope. This would only be loaded for testing purposes and avoid parsing the file multiple times on different `contractAlias`.

Reply:
Each call [was returning the singleton](https://github.com/safe-global/safe-client-gateway/pull/1195/commits/f30d5ba9b9b4bfcd7ee92e7e7cfa9930c275ef79#diff-8f7da0203c50595cd5516d468648a8a1d77aa9cb26e979339ad55c0429c01c1eR66-R72) so parsing only happened once. Nonetheless, it is instead called globally as of 83a4669872b91a720529fb3afb8608a35835f4f3.